jsonian.el

```diff
@@ -1495,35 +1497,49 @@ JSON font lock syntactic face function."
 (defun jsonian--infer-indentation ()
   "Infer the level of indentation at point."
   (save-excursion
-    ;; We examine the line above ours, under the assumption that it is correctly
-    ;; formatted.
-    (forward-line -1)
-    (jsonian--forward-to-significant-char)
-    ;; Find the column we start at
-    (let* ((start (current-column))
-           ;; FInd another column
-           (end (if (jsonian-enclosing-item)
-                    ;; We found an enclosing item, look at its column column
-                    (if (= (current-column) start)
-                        ;; The same as the starting column, We did a jump like from 1 to 2.
-                        ;;   {
-                        ;; -2->"foo": {
-                        ;;       "bar": 3
-                        ;; -1->}
-                        ;;   }
-                        ;; We repeat from 2, ensuring we will return the result of the
-                        ;; next call to `jsonian--infer-indentation'.
-                        (+ (jsonian--infer-indentation)
-                           (current-column))
-                      ;; Not the same as the starting column. This is what we want.
-                      (current-column))
-                  ;; We are not in an item, so we must be at the top level of the document.
-                  ;; Move into the document and see the indentation level of the first item.
-                  (forward-char)
-                  (jsonian--enter-collection)
-                  (jsonian--forward-to-significant-char)
-                  (current-column))))
-      (- (max start end) (min start end)))))
+    (forward-line 0)
+    (let ((indent nil)
+          (origin (point))
+          (done nil)
```

Comment:
We should be consistent with how we declare variables. `parent-position` is declared (defaulted to `nil`), we should do the same with `indent` and `done`. This is constant with the rest of the code base.
```suggestion
    (let (indent
          (origin (point))
          done
```

Reply:
My intention is that if a variable isn't initialized at declaration but needs to be initialized later, I declare it as a bare identifier.  Otherwise, i.e. if a variable is initialized to nil-as-false or nil-as-a-not-found-marker, I explicitly initialize it to nil.  This is just a quirk of mine growing up as a Java programmer.  I will follow whatever styles you prefer.